js_modules/dagit/packages/core/src/ui/Filters/FilterDropdown.tsx

```diff
@@ -261,6 +349,45 @@ const TextInputWrapper = styled.div`
   }
 `;
 
+type FilterDropdownMenuItemProps = React.ComponentProps<typeof MenuItem> & {
+  menuKey: string;
+  index: number;
+};
+const FilterDropdownMenuItem = React.memo(
+  ({menuKey, index, ...rest}: FilterDropdownMenuItemProps) => {
+    const divRef = React.useRef<HTMLDivElement | null>(null);
+    const isFocused = useRecoilValue(
+      isFocusedSelector(React.useMemo(() => ({key: menuKey, index}), [index, menuKey])),
+    );
+    React.useLayoutEffect(() => {
+      if (isFocused) {
+        divRef.current?.querySelector('a')?.focus();
+      }
+    }, [isFocused]);
+    const [hovered, setHovered] = React.useState(false);
+    return (
+      <div
+        ref={divRef}
+        onMouseEnter={() => {
+          setHovered(true);
+        }}
+        onMouseLeave={() => {
+          setHovered(false);
+        }}
```

Comment:
This visually merges the "selected" state and the "hover" state, which we don't do on menu items elsewhere and its a bit confusing. (https://dagit-storybook.vercel.app/?path=/story/suggest--default)

Reply:
Weird, I wonder if @braunjj finds it confusing too. I actually liked it better which is why I added it but I can remove it if nobody else agrees.